modules/ROOT/pages/_partials/sync/js-client-advanced-topics.adoc

```diff
@@ -1,51 +1,67 @@
-= Advanced topics
+= Advanced Topics
 
-== Implementing Custom Network Status checks
+== Network Status
 
-To use your own custom network checks, implement the https://github.com/aerogear/aerogear-js-sdk/blob/master/packages/sync/src/offline/NetworkStatus.ts[NetworkStatus]
- interface which provides two functions;
+The link:https://github.com/aerogear/aerogear-js-sdk/blob/master/packages/sync/src/offline/NetworkStatus.ts[NetworkStatus] interface can be used to check the current network status, or register a listener to perform certain actions when the status of the network changes.
 
-[source,javascript]
-----
-  onStatusChangeListener(callback: NetworkStatusChangeCallback): void;
+Two default implementations are provided to support link:https://github.com/aerogear/aerogear-js-sdk/blob/master/packages/sync/src/offline/WebNetworkStatus.ts[web browsers] and link:https://github.com/aerogear/aerogear-js-sdk/blob/master/packages/sync/src/offline/CordovaNetworkStatus.ts[Cordova]. The following example demonstrates how to register a listener using `CordovaNetworkStatus`:
 
-  isOffline(): boolean;
+[source, javascript]
 ----
 
-== Logging debug messages
+import { CordovaNetworkStatus, NetworkInfo } from '@aerogear/voyager-client';
+const networkStatus = new CordovaNetworkStatus();
+
+networkStatus.onStatusChangeListener({
+  onStatusChange: info => {
+    const online = info.online;
+    if (online) {
+      //client is online, perform some actions
+    } else {
+      //client is offline
+    }
+  }
+});
+
+let config = {
+  ...
+  networkStatus: networkStatus,
+  ...
+};
+
+//create a new client using the config
+----
 
-Sync package is using debug package to print out debug messages
+== Logging Debug Messages
 
-To enable debug please execute in console
-`localStorage.debug = 'AeroGearSync:*'`
+The {sync-client-sdk} uses the link:https://www.npmjs.com/package/debug[debug module] to log debug messages.
```

Comment:
We already have a var {sync-client} defined. sdk will need to be added if we are going with that. noticed they are both called in different places

Reply:
yeah, good catch. I will update this.